Fix TypeError on round(self.humidity) (fixes #13116)

mvn23 opened the pull request:
During init, some (most?) platforms postpone the first update for a while. This produces errors in the log as home assistant is trying to round() uninitialised variables. This commit fixes that

## Description:
fixes #13116

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`.

mvn23 committed:
Fix TypeError on round(self.humidity)

Some weather platforms postpone the first data fetch for a while on init. As a result round(self.humidity is called before it is assigned a value, producing an error. This is a fix for that.

homeassistant commented:
Hi @mvn23, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#14174).

Once you do that we will be able to review and accept this pull request.

Thanks!

OttoWinter commented on homeassistant/components/weather/__init__.py:
```diff
@@ -110,7 +110,8 @@ def state_attributes(self):
                 self.hass, self.temperature, self.temperature_unit,
                 self.precision),
-            ATTR_WEATHER_HUMIDITY: round(self.humidity)
+            ATTR_WEATHER_HUMIDITY: (round(self.humidity) if self.humidity
```
The `if self.humidity` check will not work with falsy values like `0.0`.

Write something like `if self.humidity is not None` or, even better, make it an additional check like the other attributes:

```python
data = {
            ATTR_WEATHER_TEMPERATURE: show_temp(
                self.hass, self.temperature, self.temperature_unit,
                self.precision),
        }

        humidity = self.humidity
        if humidity is not None:
            data[ATTR_WEATHER_HUMIDITY] = humidity
```

mvn23 replied:
You are absolutely right, hadn't thought about that.
Even though humidity is probably never going to be 0, we should definitely take it into account.
Fixed in the commit below.

OttoWinter reviewed (commented):
Nice 👍 Just a quick comment about falsy values.

mvn23 committed:
Rewrite to avoid false negative evaluation

As per the suggestion from @OttoWinter, rewrite to avoid matching e.g. 0.0 as false.
